pom.xml

```diff
@@ -1203,6 +1305,402 @@
+        <vers.flume>${vers.hdp.flume}</vers.flume>
+        <vers.guava>${vers.hdp.guava}</vers.guava>
+        <vers.hadoop-base>${vers.hdp.hadoop-base}</vers.hadoop-base> <!-- The version CDH is based on -->
+        <vers.hadoop>${vers.hdp.hadoop}</vers.hadoop>
+        <vers.hbase>${vers.hdp.hbase}</vers.hbase>
+        <vers.hive>${vers.hdp.hive}</vers.hive>
+        <vers.jetty>${vers.hdp.jetty}</vers.jetty>
+        <vers.jexl>${vers.hdp.jexl}</vers.jexl>
+        <vers.compress>${vers.hdp.compress}</vers.compress>
+        <vers.junit>${vers.hdp.junit}</vers.junit>
+        <vers.log4j>${vers.hdp.log4j}</vers.log4j>
+        <vers.maven>${vers.hdp.maven}</vers.maven>
+        <vers.mockito>${vers.hdp.mockito}</vers.mockito>
+        <vers.mojo-executor>${vers.hdp.mojo-executor}</vers.mojo-executor>
+        <vers.oozie>${vers.hdp.oozie}</vers.oozie> <!-- OOZIE-1790 -->
+        <vers.parquet>${vers.hdp.parquet}</vers.parquet>
+        <vers.plexus-utils>${vers.hdp.plexus-utils}</vers.plexus-utils>
+        <vers.rat>${vers.hdp.rat}</vers.rat>
+        <vers.slf4j>${vers.hdp.slf4j}</vers.slf4j>
+        <vers.codahale.metrics>${vers.hdp.codahale.metrics}</vers.codahale.metrics>
+        <vers.typesafe.config>${vers.hdp.typesafe.config}</vers.typesafe.config>
+        <vers.solr>${vers.hdp.solr}</vers.solr>
+        <solr.expected.version>${solr.hdp.expected.version}</solr.expected.version> <!-- sanity check to verify we actually run against the expected version rather than some outdated version -->
+        <vers.tika>${vers.hdp.tika}</vers.tika>
+        <vers.jackson>${vers.hdp.jackson}</vers.jackson>
+        <vers.opencsv>${vers.hdp.opencsv}</vers.opencsv>
+        <vers.jodatime>${vers.hdp.jodatime}</vers.jodatime>
+        <vers.jcommander>${vers.hdp.jcommander}</vers.jcommander>
+        <vers.spark>${vers.hdp.spark}</vers.spark>
+        <vers.snappy>${vers.hdp.snappy}</vers.snappy>
+      </properties>
+      <build>
+        <pluginManagement>
+          <plugins>
+            <!--plugin>
+              <groupId>org.apache.maven.plugins</groupId>
+              <artifactId>maven-jar-plugin</artifactId>
+              <version>${vers.maven-jar-plugin}</version>
+              <configuration>
+                <classifier>hdp${hdp2.version}</classifier>
+              </configuration>
+            </plugin-->
+            <plugin>
+              <groupId>org.apache.maven.plugins</groupId>
+              <artifactId>maven-shade-plugin</artifactId>
+              <version>${vers.maven-shade-plugin}</version>
+              <executions>
+                <execution>
+                  <phase>package</phase>
+                  <goals>
+                    <goal>shade</goal>
+                  </goals>
+                  <configuration>
+                    <artifactSet>
+                      <includes>
+                        <include>com.google.guava:guava</include>
```

Comment:
Yes, Kite shades guava in every artifact (which kinda sucks). When you do that, you want to reference the real package names and then let shading re-write your access when the final jar is assembled. IMO, Kit would have been better served to create a single shaded artifact and to use shaded imports everywhere else to avoid duplicating guava everywhere. It would have saved me the trouble of doing `maven-install-plugin` and `maven-deploy-plugin` tricks everywhere.